### ISSUE TITLE
Querying can get really slow in large data sets

### ISSUE THREAD

**Issue by qexpres** [realm/realm-studio#1084]
**Is your feature request related to a problem? Please describe.**
When querying in classes with a large amount of objects, Studio is quite slow.

**Describe the solution you'd like**
I think it has to do with the fact that it evaluates the query with each key stroke. My suggestion would be to only evaluate the query after the user stopped typing. You could use key stroke delays or require the user to submit the query using a button or the return key.

**Describe alternatives you've considered**
There really is no alternative, is there?

**Additional context**
-

**Comment by kraenhansen** [comment]
Thanks for reporting this. How many objects is a large amount of objects in your experience? And could I get you to share what OS your using Studio on?

**Comment by qexpres** [comment]
Of course, I'm running Realm Studio (3.3.0.546) on Mac OS Mojave (10.14.2). I think I started using Realm Studio from version 3.0 and the behaviour I described has always been te same.

Having a collection of ~100 objects causes delays of a few 100 ms every time I press a key in the query bar. A collection of ~250,000 objects causes delays up to a second. It doesn't have to be a valid query for the delays to occur. The delays seem to be related to the result of the query as well; smaller query results will reduce the delay time when changing the query until you get a bigger result again.

Maybe the lag is caused by the application still drawing ~250,000 empty rows. Which is ok, as long as it doesn't happen with every keystroke. A quick win would be to not redraw view for invalid queries if it's UI related at all.
